File: cwclock-mobile/src/components/ErrorBanner.js
import React from "react";
import { StyleSheet, Text } from "react-native";
import theme from "../theme";

const ErrorBanner = ({ message }) => {
  if (!message) return null;
  return <Text style={styles.text}>{message}</Text>;
};

const styles = StyleSheet.create({
  text: {
    color: theme.color.danger,
    marginBottom: theme.spacing(1.5),
    fontSize: 14,
  },
});

export default ErrorBanner;

File: cwclock-mobile/lib/widgets/error_banner.dart
import 'package:flutter/material.dart';

import '../theme.dart';

/// Ported from src/components/ErrorBanner.js.
class ErrorBanner extends StatelessWidget {
  final String? message;

  const ErrorBanner({super.key, this.message});

  @override
  Widget build(BuildContext context) {
    if (message == null || message!.isEmpty) return const SizedBox.shrink();
    return Padding(
      padding: EdgeInsets.only(bottom: AppSpacing.of(1.5)),
      child: Text(message!, style: TextStyle(color: AppColors.of(context).danger, fontSize: 14)),
    );
  }
}
